### PR TITLE
Add "error handling" java docs section to from callable & co

### LINKED ISSUE #6179: Add an "Error handling" JavaDocs section to fromCallable & co
The operators of `fromCallable`, `fromAction` and `fromRunnable` in the reactive classes could use an `<dt><b>Error handling:</b></dt>` section in the JavaDocs describing the regular and canceled/disposed behavior.

> If the `{@link Callable/Action/Runnable}` throws an exception/exception/unchecked exception, the respective `{@link Throwable}` is delivered to the downstream via `{@link Subscriber/Observer/.../#onError()}`, except when the downstream has canceled/disposed this `{@code Flowable/Observable/Single/Maybe/Completable}` source. In this latter case, the Throwable is delivered to the global error handler via `{@link RxJavaPlugins#onError(Throwable)}` as `{@link io.reactivex.exceptions.UndeliverableException UndeliverableException}`.

### PR DESCRIPTION
**Pull request by luis-cortes**
Resolves #6179

Added JavaDocs describing the regular and canceled/disposed behavior for

- `Observable.fromCallable()`
- `Single.fromCallable()`
- `Maybe.fromAction()`
- `Completable.fromCallable()`
- `Completable.fromAction()`
- `Flowable.fromCallable()`

Did not alter `Maybe.fromCallable()` because the existing JavaDoc appears to be adequate.

### CODE REVIEW

File: src/main/java/io/reactivex/Completable.java
Comment (akarnokd) [inline src/main/java/io/reactivex/Completable.java]: "downstream has disposed this" as only `Flowable`s have `cancel`.

File: src/main/java/io/reactivex/Completable.java
Comment (akarnokd) [inline src/main/java/io/reactivex/Completable.java]: "downstream has disposed this"

File: src/main/java/io/reactivex/Maybe.java
Comment (akarnokd) [inline src/main/java/io/reactivex/Maybe.java]: "downstream has disposed this"

File: src/main/java/io/reactivex/Observable.java
Comment (akarnokd) [inline src/main/java/io/reactivex/Observable.java]: "downstream has disposed this"

File: src/main/java/io/reactivex/Single.java
Comment (akarnokd) [inline src/main/java/io/reactivex/Single.java]: "downstream has disposed this"

**Review by akarnokd — COMMENTED** [review]
Small adjustments would be great. Could you also update `Flowable.fromCallable`?

**Review by akarnokd — APPROVED** [review]

**Comment by luis-cortes** [comment]
I noticed that there's quite a bit more of the "cancellation" language in the other areas. 
For example: 
- Single.java:833 `...and all other source {@code SingleSource}s are cancelled.`
- Completable.java:1588 ` * @param onFinally the action called when this Completable terminates or gets cancelled`

Should this be added to #4535?

**Comment by akarnokd** [comment]
Post a separate PR for those.

**Comment by luis-cortes** [comment]
Should I create a separate issue as well? 

Also, is there anything else you need from me to get this PR merged (unsure of the process at this point)?

**Comment by akarnokd** [comment]
No need to open an issue for that.

We are waiting for another approval and this will be merged.

**Review by vanniktech — COMMENTED** [review]
:+1:  missed it

**Pull request merged**
